src/solver/conv_hip_implicit_gemm_fwd_v4r5_xdlops.cpp

```diff
@@ -0,0 +1,1052 @@
+
+        if(data_per_thread_copy_b * NWaves == 0 ||
+           data_per_thread_copy % (data_per_thread_copy_b * NWaves) != 0)
+            MIOPEN_THROW("invalid performance parameter");
+
+        const auto tmp = data_per_thread_copy / (data_per_thread_copy_b * NWaves);
+
+        int data_per_thread_copy_gemmkpack = -1;
+        int data_per_thread_copy_gemmk     = -1;
+        if(GemmBThreadCopyMoreGemmKPack)
+        {
+            data_per_thread_copy_gemmkpack = gcd(GemmKPack, tmp);
+            data_per_thread_copy_gemmk     = tmp / data_per_thread_copy_gemmkpack;
+        }
+        else
+        {
+            data_per_thread_copy_gemmk     = gcd(GemmKPerBlock, tmp);
+            data_per_thread_copy_gemmkpack = tmp / data_per_thread_copy_gemmk;
+        }
+
+        // vector write into LDS
+        DstDataPerWrite_GemmKPack = gcd(DstDataPerWrite_GemmKPack, data_per_thread_copy_gemmkpack);
+
+        if(!(data_per_thread_copy_gemmkpack > 0 && data_per_thread_copy_gemmk > 0 &&
+             data_per_thread_copy_b > 0))
+        {
+            MIOPEN_THROW("invalid performance parameter");
+        }
+
+        if(!(GemmKPerBlock % data_per_thread_copy_gemmk == 0 &&
+             BPerBlock % data_per_thread_copy_b == 0 &&
+             GemmKPack % data_per_thread_copy_gemmkpack == 0))
+            MIOPEN_THROW("invalid performance parameter");
+
+        ClusterLengths_GemmK     = GemmKPerBlock / data_per_thread_copy_gemmk;
+        ClusterLengths_B         = BPerBlock / data_per_thread_copy_b;
+        ClusterLengths_GemmKPack = GemmKPack / data_per_thread_copy_gemmkpack;
+
+        // blockwise-copy support that block_size is larger than thread cluster size, which means
+        // some threads may not do threadwise copy
+        if(block_size != ClusterLengths_GemmK * ClusterLengths_B * ClusterLengths_GemmKPack)
+            MIOPEN_THROW("invalid performance parameter");
+    }
+    catch(...)
+    {
+        return std::make_tuple(-1, -1, -1, -1, -1, false);
+    }
+
+    return std::make_tuple(ClusterLengths_GemmK,
+                           ClusterLengths_B,
+                           ClusterLengths_GemmKPack,
+                           SrcDataPerRead_B,
+                           DstDataPerWrite_GemmKPack,
+                           true);
+}
+
+std::tuple<std::size_t, bool> PerformanceImplicitGemmForwardV4R5Xdlops::CalculateLdsNumberOfByte(
+    const ConvolutionContext& ctx) const
+{
+    const auto a_block_space = GemmKPerBlock * GemmMPerBlock * GemmKPack;
+    const auto b_block_space = GemmKPerBlock * GemmNPerBlock * GemmKPack;
+
+    std::size_t lds_size =
+        (a_block_space + b_block_space) * (ctx.IsFp32() ? sizeof(float) : sizeof(half_float::half));
+
+    return std::make_tuple(lds_size, true);
+}
+
+// Used by IsReallyValid()
+bool PerformanceImplicitGemmForwardV4R5Xdlops::IsValidValue() const
+{
+    // clang-format off
+    return IsTwoPower<4, 256>(GemmMPerBlock)
```

Comment:
this function should verify GemmBThreadDataPerRead_GemmN ？

Reply:
Fixed